### PR TITLE
make full text search on native currency name & symbol

### LINKED ISSUE #951: Token search
**Description**
There is a bug in token search:
If you type "MA" in search "MATIC" appears (and it's correct behaviour)
But if you try search from "the middle" for example typing "TIC" MATIC will not appear (but wrapped MATIC will...)

**How to Reproduce**
1. Go to Swap page
2. Click on "Select token" on Swap box
3. Then type "MATIC" in a search bar
4. Then type "TIC"  in a search bar

**Screenshots**
image
image

### PR DESCRIPTION
**Pull request by denciu**
# Summary

Closes #951 

Native currency is now included in search results on similar rules as regular token (not only when query is beginning of native currency symbol).

To test:
1. Open token search in any place (swap/bridge) on any chain (examples are on mainnet as symbol is different than name (ETH, Ether)
2. Following queries should include ETH on token list (should work even when query starts with spaces):
- eth
- er
- the
- ...etc

### CODE REVIEW

**Review by MilanVojnovic95 — APPROVED** [review]
Tested and now it works great! 👍

**Review by mix1o — APPROVED** [review]

**Review by wixzi — APPROVED** [review]

File: src/components/SearchModal/CurrencySearch/CurrencySearch.component.tsx [hunk 1]
```diff
@@ -72,11 +72,12 @@ export const CurrencySearch = ({
...
 
   const filteredSortedTokensWithNativeCurrency: Currency[] = useMemo(() => {
-    if (!showNativeCurrency) return filteredSortedTokens
-    const s = debouncedQuery.toLowerCase().trim()
-    if (nativeCurrency.symbol && nativeCurrency.symbol.toLowerCase().startsWith(s)) {
+    if (!showNativeCurrency || !nativeCurrency.symbol || !nativeCurrency.name) return filteredSortedTokens
+
+    if (new RegExp(debouncedQuery.replace(/\s/g, ''), 'gi').test(`${nativeCurrency.symbol} ${nativeCurrency.name}`)) {
       return nativeCurrency ? [nativeCurrency, ...filteredSortedTokens] : filteredSortedTokens
     }
```
Comment (wixzi) [inline src/components/SearchModal/CurrencySearch/CurrencySearch.component.tsx:79]:
```suggestion
    if (nativeCurrency  && (new RegExp(debouncedQuery.replace(/\s/g, ''), 'gi').test(`${nativeCurrency.symbol} ${nativeCurrency.name}`)) {
      return [nativeCurrency, ...filteredSortedTokens]
    }
```

Maybe in a single check :)

**Review by wixzi — APPROVED** [review]

**Comment by niemam29** [comment]
Hey! I've noticed that there is some problem when searching for xdai in bridge (gnosis network)

https://user-images.githubusercontent.com/30408272/173299355-da554741-d073-4486-aa4a-0173441d63ee.mp4

**Comment by denciu** [comment]
> Hey! I've noticed that there is some problem when searching for xdai in bridge (gnosis network)
> 
>  s.mp4

For some reason he didn't show token list if only token available was native currency. Fixed in 

> Hey! I've noticed that there is some problem when searching for xdai in bridge (gnosis network)
> 
>  s.mp4

Fixed in [669f1e6](https://github.com/levelkdev/swapr-dapp/pull/1074/commits/669f1e6fc3c80e8f109431c61d0eee0f8a084ebf)

**Comment by MilanVojnovic95** [comment]
Tested and it looks like the bug is no longer present, looks ok to me 👍

**Pull request merged**